### PR TITLE
Cleaned up rubocop warnings and updated CONTRIBUTING and ISSUES docs.

### PR DESCRIPTION
**Pull request by rayrod2030**
This pull request changes the following:

* Cleaned up all Rubocop warnings.
* Updated test kitchen platforms to support ubuntu 12.04 and 14.04 and removed ubuntu 13.04 which is EOL.
* Updated EC2 AMI's to more recent builds.
* Removed gemset hint files.
* Added new CONTRIBUTING and ISSUES markdown.
* Updated README to link to those new files.

### CODE REVIEW

File: .kitchen.ec2.yml [hunk 1]
```diff
@@ -1,69 +1,49 @@
...
-  aws_ssh_key_id: <%= ENV['AWS_SSH_KEY_ID'] || 'default' %>
-  ssh_key: <%= File.expand_path(ENV['AWS_SSH_KEY_PATH'] ||
-                             "#{ENV['HOME']}/.ssh/id_rsa") %>
+  aws_access_key_id: <%= AWS_ACCESS_KEY_ID %>
+  aws_secret_access_key: <%= AWS_ACCESS_KEY_SECRET %>
+  aws_ssh_key_id: <%= AWS_SSH_KEY_ID %>
+  ssh_key: <%= AWS_SSH_KEYPATH %>
   region: us-east-1
   availability_zone: us-east-1c
   flavor_id: <%= ENV['AWS_FLAVOR_ID'] || '(ENTER_AWS_FLAVOR_ID)' %>
```
Comment (mjcdiggity) [inline .kitchen.ec2.yml]: Why is this here vs in above section containing constants?

File: .kitchen.ec2.yml [hunk 2]
```diff
@@ -1,69 +1,49 @@
...
+  require_chef_omnibus: true
   tags:
     created-by: 'test-kitchen'
     launched-by: <%= ENV['USER'] || '(ENTER_USER)' %>
```
Comment (mjcdiggity) [inline .kitchen.ec2.yml]: Same as above.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,36 +1,69 @@
...
+Please also keep in mind:
+
+- Be patient as not all items will be tested or reviewed immediately by the core
+team.
+- Be receptive and responsive to feedback about your additions or changes. The
+core team and/or other community members may make suggestions or ask questions
+about your change. This is part of the review process, and helps everyone to
+understand what is happening, why it is happening, and potentially optimizes
+your code.
+- Be understanding
```
Comment (mjcdiggity) [inline CONTRIBUTING.md]: `.`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,36 +1,69 @@
...
+If you're looking to contribute but aren't sure where to start, check out the
+open issues.
+
+
+Will Not Merge
+--------------
+This second details Pull Requests that we will **not** merge.
+
+1. New features without accompanying Test Kitchen tests
+1. New features without accompanying usage documentation
```
Comment (mjcdiggity) [inline CONTRIBUTING.md]: Can this be changed to 2., although it does show up correctly in markdown.

File: .kitchen.ec2.yml (hunk 1 above)
Comment (rayrod2030) [inline .kitchen.ec2.yml]: This is where these values are actually passed to test-kitchen.

Comment (mjcdiggity) [inline .kitchen.ec2.yml]: I'm saying all key values are read from environment variables into constants in block at top of file, but not flavor_id/user. Unless those are prompted for when you run test-kitchen? Just looking for consistency if they are ultimately treated the same way.

File: CONTRIBUTING.md (hunk 4 above)
Comment (rayrod2030) [inline CONTRIBUTING.md]: Pro Tip:  If you number every numbered item with 1. markdown takes care of the actual order.  Saves time later if you add or remove items.

**Comment by mjcdiggity** [comment]
Everything looks good, will merge once comments are resolved.

File: CONTRIBUTING.md (hunk 4 above)
Comment (mjcdiggity) [inline CONTRIBUTING.md]: Pro tip indeed :+1:

Comment (mjcdiggity) [inline CONTRIBUTING.md]: TIL
